tests/v2/validation/charts/backup_restore/backup_restore_test.go

```diff
@@ -0,0 +1,204 @@
+	require.NoError(b.T(), err)
+
+	logrus.Info("Creating two users...")
+	u1, err := users.CreateUserWithRole(client, users.UserConfig(), member)
+	require.NoError(b.T(), err)
+	userList = append(userList, u1)
+
+	u2, err := users.CreateUserWithRole(client, users.UserConfig(), member)
+	require.NoError(b.T(), err)
+	userList = append(userList, u2)
+
+	logrus.Info("Creating two projects...")
+	p1, err := createProject(client, b.project.ClusterID)
+	require.NoError(b.T(), err)
+	projList = append(projList, p1)
+
+	p2, err := createProject(client, b.project.ClusterID)
+	require.NoError(b.T(), err)
+	projList = append(projList, p2)
+
+	logrus.Info("Creating two role templates...")
+	rt1, err := createRole(client, roleContext, roleName, rules)
+	require.NoError(b.T(), err)
+	roleList = append(roleList, rt1)
+
+	rt2, err := createRole(client, roleContext, roleName, rules)
+	require.NoError(b.T(), err)
+	roleList = append(roleList, rt2)
+
+	logrus.Info("Creating a backup of the local cluster...")
+	firstBackup, err := createBackup(client, backupName)
+	require.NoError(b.T(), err)
+	charts.VerifyBackupCompleted(b.T(), client, backupSteveType, firstBackup)
+
+	backupObj, err := client.Steve.SteveType(backupSteveType).ByID(backupName)
+	require.NoError(b.T(), err)
+	backupK8Obj := &bv1.Backup{}
+	error := v1.ConvertToK8sType(backupObj.JSONResp, backupK8Obj)
+	require.NoError(b.T(), error)
+	backupFileName := backupK8Obj.Status.Filename
+
+	logrus.Info("Validating backup file is in AWS S3...")
+	backupPresent, err := validateAWSS3BackupExists(b.T(), broConfig.S3BucketName, broConfig.S3FolderName, backupFileName)
+	require.NoError(b.T(), err)
+	assert.True(b.T(), backupPresent)
+
+	logrus.Info("Creating two more users...")
+	u3, err := users.CreateUserWithRole(client, users.UserConfig(), member)
+	require.NoError(b.T(), err)
+	userList = append(userList, u3)
+
+	u4, err := users.CreateUserWithRole(client, users.UserConfig(), member)
+	require.NoError(b.T(), err)
+	userList = append(userList, u4)
+
+	logrus.Info("Creating two more projects...")
+	p3, err := createProject(client, b.project.ClusterID)
+	require.NoError(b.T(), err)
+	projList = append(projList, p3)
+
+	p4, err := createProject(client, b.project.ClusterID)
+	require.NoError(b.T(), err)
+	projList = append(projList, p4)
+
+	logrus.Info("Creating two more role templates...")
+	rt3, err := createRole(client, roleContext, roleName, rules)
+	require.NoError(b.T(), err)
+	roleList = append(roleList, rt3)
+
+	rt4, err := createRole(client, roleContext, roleName, rules)
+	require.NoError(b.T(), err)
+	roleList = append(roleList, rt4)
+
+	logrus.Infof("Creating a restore using backup file: %v", backupFileName)
+	completedRestore, err := createRestore(client, backupFileName)
+	require.NoError(b.T(), err)
+	charts.VerifyRestoreCompleted(b.T(), client, restoreSteveType, completedRestore)
+
+	logrus.Info("Restore complete, validating Rancher resources...")
+	verifyUserResources(b.T(), client, userList)
```

Comment:
we can consolidate three of the user resource validation to one function such that it accepts rolesList, projectList and userList and can just call verifyResources()